src/deluge/gui/views/session_view.cpp

```diff
@@ -709,6 +709,7 @@ ActionResult SessionView::padAction(int32_t xDisplay, int32_t yDisplay, int32_t
 
 				// If Clip present here...
 				else if (clip) {
+					currentSong->currentClip = clip;
```

Comment:
I think this belongs around line 734 and 778, where selectedClipYDisplay is set - otherwise it could select a clip that's being deleted